src/main/java/com/aws/greengrass/logmanager/LogManagerService.java

```diff
@@ -579,12 +595,15 @@ private void processLogsAndUpload() throws InterruptedException {
                     }
                     // Sort the files by the last modified time.
                     allFiles.sort(Comparator.comparingLong(File::lastModified));
-                    // If there are no rotated log files for the component, then return.
-                    if (allFiles.size() - 1 <= 0) {
-                        continue;
-                    }
-                    // Don't consider the active log file.
+
+                     // sublist allFiles to rotated file and active file, calculate the hash for active file with
+                    // some lines, default is 1 line
                     allFiles = allFiles.subList(0, allFiles.size() - 1);
```

Comment:
`allFiles` is already excluding the active log file. Your added code is further slicing a list which has already had the active file removed